core/src/main/clojure/xtdb/pgwire.clj

```diff
@@ -1032,7 +1088,7 @@
 
                      (let [^RelationReader args-rel (aget !args 0)]
                        (case (:statement-type inner)
-                         :query (util/with-close-on-catch [inner-cursor (.openQuery inner-pq (assoc query-opts :args args-rel))]
+                         :query (util/with-close-on-catch [inner-cursor (with-auth-check conn (.openQuery inner-pq (assoc query-opts :args args-rel)))]
```

Comment:
same here (although I know that's the first thing) - this one is more about making it easier to spot rather than hiding it under a binding